AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -0,0 +1,116 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Object to house all the Qt Objects used when testing and manipulating the O3DE UI
+"""
+from editor_python_test_tools.utils import TestHelper as helper
+from PySide2 import QtWidgets, QtCore
+import pyside_utils
+from types import SimpleNamespace
+from consts.scripting import (VARIABLE_MANAGER_QT, VARIABLE_PALETTE_QT, ADD_BUTTON_QT, GRAPH_VARIABLES_QT)
+from consts.general import (WAIT_TIME_3)
+from editor_python_test_tools.utils import Report
+
+class Tests():
+    variable_count_expected = ("Variable count matches expected", "Variable count does not match expected")
+
+"""
+Basic variable types.
+These are str-str kv pairings so they can be used with SimpleNamespace class easily or as an iterable type.  
+"""
+VARIABLE_TYPES_DICT = {
+    "Boolean": "Boolean",
+    "Color": "Color",
+    "EntityId": "EntityId",
+    "Number": "Number",
+    "String": "String",
+    "Transform": "Transform",
+    "Vector2": "Vector2",
+    "Vector3": "Vector3",
+    "Vector4": "Vector4"
+}
+
+class QtPyScriptCanvasVariableManager:
+    """
+    QtPy class for handling the behavior of the script canvas variable manager
+    """
+
+    def __init__(self, sc_editor):
+        self.variable_manager = sc_editor.sc_editor.findChild(QtWidgets.QDockWidget, VARIABLE_MANAGER_QT)
+
+    def create_new_variable(self, new_variable_type):
+        """
+        function for adding a new variable to the variable manager's list
+
+        param new_variable_type: the type of variable you want to create provided as a string. Is case-sensitive!
+
+        returns: None
+        """
+        self.__validate_new_variable(new_variable_type)
+
+        add_new_variable_button = self.variable_manager.findChild(QtWidgets.QPushButton, ADD_BUTTON_QT)
+        add_new_variable_button.click()  # Click on Create Variable button
+
+        helper.wait_for_condition((
+            lambda: self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT) is not None), WAIT_TIME_3)
+
+        # Select variable type
+        table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
+        model_index = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
+
+        # Click on it to create variable
+        pyside_utils.item_view_index_mouse_click(table_view, model_index)
+
+    def __validate_new_variable(self, new_variable_type):
```

Comment:
Since this is a "private" helper function, I'd move this to the top (below init) or bottom of the class instead of mixed in.

Reply:
moved the function up in the script file.